src/customPromptProcessor.ts

```diff
@@ -19,33 +19,29 @@ export interface CustomPrompt {
 
 export class CustomPromptProcessor {
   private static instance: CustomPromptProcessor;
+  private usageStrategy: TimestampUsageStrategy;
 
-  private constructor(
-    private vault: Vault,
-    private settings: CopilotSettings,
-    private usageStrategy?: PromptUsageStrategy
-  ) {}
-
-  static getInstance(
-    vault: Vault,
-    settings: CopilotSettings,
-    usageStrategy?: PromptUsageStrategy
-  ): CustomPromptProcessor {
+  private constructor(private vault: Vault) {
+    this.usageStrategy = new TimestampUsageStrategy();
+  }
+
+  get customPromptsFolder(): string {
+    return getSettings().customPromptsFolder;
+  }
+
+  static getInstance(vault: Vault): CustomPromptProcessor {
     if (!CustomPromptProcessor.instance) {
-      if (!usageStrategy) {
-        console.warn("PromptUsageStrategy not initialize");
-      }
```

Comment:
This code was very fragile. Many callers do not have access a usage strategy instance. If they ever get called before main.ts, it will not work. With the settings refactoring, this class can instantiate its usageStrategy now.